components-2/readme.md

```diff
@@ -0,0 +1,166 @@
+- [ ] 🪄 Create a component on a web page
+- [ ] 📖 Write a function to build a component from a documented pattern
+- [ ] 👪🏿 Collaborate to build a page
+```
+
+<!--{{<note type="activity" title="Set up your working branch 15m">}}-->
+
+### Set up your working branch
+
+Everyone should do this so everyone is set up correctly.
+
+1. You will be working on a special branch called [FOR-USE-IN-CLASS](https://github.com/CodeYourFuture/CYF-Workshops/tree/FOR-USE-IN-CLASS).
+1. First, **clone** [the workshops repository](https://github.com/CodeYourFuture/CYF-Workshops/)
+1. Checkout `git checkout FOR-USE-IN-CLASS`
+1. Your cohort has a named _sub branch_. Everyone, check that out next with `git checkout COHORT` (eg `NW6`). This is your working branch:
+
+```mermaid
+gitGraph
+    commit "main"
+    branch FOR-USE-IN-CLASS
+    checkout FOR-USE-IN-CLASS
+    branch NW6
+    checkout NW6
+    branch feature/header
+    commit
+    commit
+    checkout NW6
+    branch feature/footer
+    commit
+    commit
+    commit
+    checkout NW6
+    branch feature/card
+    commit
+    commit
+    commit
+    checkout NW6
+    branch feature/menu
+    commit
+    commit
+    commit
+    checkout NW6
+    merge feature/header
+    merge feature/footer
+    merge feature/card
+    merge feature/menu
+```
+
+Check you are in the right place with GitLens!
+
+<!--{{</note>}}-->
+
+<!--{{<note type="tip" title="Your working branch is your cohort name">}}-->
+
+Your COHORT branch is your cohort name. It might be NW6, WM5, GL7...You're working in your COHORT branch and NOT in main.
+
+<!--{{</note>}}-->
+
+Next, divide the class into four teams. Use the [team generator](teams.html) to help you do this.
+
+Set a whole class [timer for 10 minutes](https://www.google.com/search?q=10+minute+time).
+
+### Each team, choose a driver
+
+1. Driver, ask everyone in your team to confirm you are in the right working branch.
+1. Driver, from this branch, create and checkout a new branch for your team named like this: `feature/component`, where _cohort_ is your cohort name and _component_ is the component you're building, for example: `feature/header`
+
+```bash
+git switch --create feature/cohort-component
+```
+
+Now you're all set up to make changes to together.html.
+
+<!--{{</note>}}-->
+
+<!--{{<note type="activity" title="Find your docs 10m">}}-->
+
+## Find your docs
+
+Set a whole class [timer for 10 minutes](https://www.google.com/search?q=10+minute+time).
+
+Deployed from this repo is a [mini design system](https://cyf-workshops.netlify.app/components-2). This design system has several documented components, and some also have javascript files.
+
+Your mission is to find the docs for each component with an associated script file, and read them. Discuss your findings. Look at the pages on the website with Devtools. Compare the Inspect view with the source code you can see in the files. How are they different? How are HTML and the DOM different?
```

Comment:
So I think the intention is that trainees figure out that they need to import the script and call the "component" - is that right?

Reply:
Right, all the pages are built in this way so they can just copy and paste it from index.html or any doc page.